### PR TITLE
Add coq-sail package

### PR DESCRIPTION
**Pull request by bacam**
This is the support library for Coq models generated by the Sail instruction set specification tool.

### CODE REVIEW

File: released/packages/coq-sail/coq-sail.0.16/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+dev-repo: "git+https://github.com/rems-project/coq-sail.git"
+build: [make "-C" "src" "HAVE_OPAM_BBV=yes"]
+install: [make "-C" "src" "install"]
+depends: [
+  "coq" {>= "8.9.0" & <= "8.17.1"}
+  "coq-bbv" {>= "1.1" & <= "1.3"}
+  # The sail package isn't strictly required, but this is the easiest way for
+  # models' opam files to ensure that the version matches
+  "sail" {= version}
+]
```
Comment (palmskog) [inline released/packages/coq-sail/coq-sail.0.16/opam:30]: It's not very CI friendly to depend on a package that's not even required for this package. How about the following instead that we use for Ott?
```suggestion
]
conflicts: [
  "sail" { != version }
]
```

File: released/packages/coq-sail/coq-sail.0.16/opam [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+engineer-friendly, vendor-pseudocode-like language for describing
+instruction semantics. It is essentially a first-order imperative
+language, but with lightweight dependent typing for numeric types and
+bitvector lengths, which are automatically checked using Z3. It has
+been used for several papers, available from
+http://www.cl.cam.ac.uk/~pes20/sail/.
+The Sail tool can be found in main opam repository."""
+tags: [
+  "logpath:Sail"
+  "category:CS/Semantics and Compilation/Semantics"
```
Comment (palmskog) [inline released/packages/coq-sail/coq-sail.0.16/opam:46]:
```suggestion
  "category:Computer Science/Semantics and Compilation/Semantics"
```

File: released/packages/coq-sail/coq-sail.0.16/opam [hunk 3]
```diff
@@ -0,0 +1,52 @@
+opam-version: "2.0"
+name: "coq-sail"
+version: "0.16"
```
Comment (palmskog) [inline released/packages/coq-sail/coq-sail.0.16/opam:3]:
```suggestion
```
This is already given by the repository `opam` file path.

Comment (bacam) [reply released/packages/coq-sail/coq-sail.0.16/opam]: Oops, an artifact of starting from the file used for pinning...

**Pull request merged**